Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -854,6 +854,44 @@ static CheckedPtr<const Element> anchorScopeForAnchorName(const RenderBoxModelOb
     return nullptr;
 }
 
+enum class TopLayerStatus : uint8_t { Same, Lower, Higher };
+static TopLayerStatus computeTopLayerStatus(const RenderBox& anchored, const RenderBoxModelObject& anchor)
+{
+    // Two elements are in the same top layer if they have the same top layer root (including if both are none).
+    // An element A is in a higher top layer than an element B if A has a top layer root, and either B has a top
+    // layer root earlier in the top layer than A’s, or B doesn’t have a top layer root at all.
+    // https://drafts.csswg.org/css-position-4/#top-layer
+
+    if (!anchored.document().hasTopLayerElement())
+        return TopLayerStatus::Same;
+
+    auto topLayerRoot = [&](auto& renderer) -> const RenderLayerModelObject* {
```

Comment:
I would call this `topLayerAncestor` / `rootInTopLayer` because `topLayerRoot` suggests it's the root of all the top layer elements (which is the RenderView or wherever we append top layer stuff nowadays)